apm-agent-plugins/apm-profiling-plugin/src/main/java/co/elastic/apm/agent/profiler/asyncprofiler/AsyncProfiler.java

```diff
@@ -64,24 +65,31 @@
     private AsyncProfiler() {
     }
 
-    public static AsyncProfiler getInstance() {
+    public static AsyncProfiler getInstance(ProfilingConfiguration config) {
```

Comment:
[very minor] just a personal preference so that the AsyncProfiler class doesn't depend on the configuration class. Feel free to leave as-is 
```suggestion
    public static AsyncProfiler getInstance(String profilerLibDirectory) {
```

Reply:
I'd um'd and ah'd between the two - this does make things a lot cleaner 👍